test/common/http/codec_wrappers_test.cc

```diff
@@ -8,29 +8,29 @@ using testing::_;
 namespace Envoy {
 namespace Http {
 
-class MockStreamEncoderWrapper : public StreamEncoderWrapper {
+class MockRequestStreamEncoderWrapper : public RequestStreamEncoderWrapper {
 public:
-  MockStreamEncoderWrapper() : StreamEncoderWrapper(inner_encoder_) {}
+  MockRequestStreamEncoderWrapper() : RequestStreamEncoderWrapper(inner_encoder_) {}
   void onEncodeComplete() override { encode_complete_ = true; }
 
   MockStreamEncoder& innerEncoder() { return inner_encoder_; }
   bool encodeComplete() const { return encode_complete_; }
 
 private:
-  MockStreamEncoder inner_encoder_;
+  MockRequestStreamEncoder inner_encoder_;
   bool encode_complete_{};
 };
 
-TEST(StreamEncoderWrapper, HeaderOnlyEncode) {
-  MockStreamEncoderWrapper wrapper;
+TEST(RequestStreamEncoderWrapper, HeaderOnlyEncode) {
+  MockRequestStreamEncoderWrapper wrapper;
 
   EXPECT_CALL(wrapper.innerEncoder(), encodeHeaders(_, true));
   wrapper.encodeHeaders(TestHeaderMapImpl{{":status", "200"}}, true);
   EXPECT_TRUE(wrapper.encodeComplete());
 }
 
-TEST(StreamEncoderWrapper, HeaderAndBodyEncode) {
-  MockStreamEncoderWrapper wrapper;
+TEST(RequestStreamEncoderWrapper, HeaderAndBodyEncode) {
+  MockRequestStreamEncoderWrapper wrapper;
```

Comment:
So I was going to give a pass on checking that we had request and response right in tests that don't care, but I'm concerned about causing confusion for people looking at tests.

I'd suggest a (temporary?) ASSERT in the mock and the production classes that the response has a :status and the request has a :method to make sure we're using the right ones in the right tests.  WDYT?

Reply:
In this change, it doesn't actually matter, as there is no actual code change. In my next change (WIP) I move headers/trailers methods, and it becomes impossible for anything to compile without using the correct interface. If it's OK with you I would probably just defer to that since it will be an immediate follow up?